tests/misc_env/subscription_manager.py

```diff
@@ -0,0 +1,151 @@
+import re
+
+from ceph.waiter import WaitUntil
+from cli.utilities.packages import (
+    Package,
+    SubscriptionManager,
+    SubscriptionManagerError,
+)
+from cli.utilities.utils import os_major_version
+from utility.log import Log
+from utility.utils import get_cephci_config
+
+log = Log(__name__)
+
+
+class ConfigNotFoundError(Exception):
+    pass
+
+
+def setup_subscription_manager(ceph, server, timeout=300, interval=60):
+    """
+    Set up the subscription manager on a Ceph node with provided credentials.
+
+    This includes:
+    - Unregistering the system if already registered
+    - Cleaning all local subscription data using `subscription-manager clean`
+    - Removing any residual certificate or identity files
+    - Registering to the specified subscription server
+
+    Args:
+        ceph: Ceph node object where subscription-manager is to be configured.
+        server: Name of the subscription server, used to fetch credentials from config.
+        timeout: Max time (in seconds) to wait for successful registration.
+        interval: Interval (in seconds) between retry attempts.
+
+    Returns:
+        bool: True if subscription is successful, False otherwise.
+
+    Raises:
+        ConfigNotFoundError: if credentials for the given server are not found.
+    """
+    configs = get_cephci_config()
+    creds = configs.get(f"{server}_credentials")
+    if not creds:
+        raise ConfigNotFoundError(f"{server} credentials are not provided")
+
+    sm = SubscriptionManager(ceph)
+
+    try:
+        sm.unregister()
+    except SubscriptionManagerError:
+        # Ignore errors like "system not registered"
+        pass
+
+    try:
+        sm.clean()
+    except SubscriptionManagerError:
+        # Ignore non-critical clean failures
+        pass
+
+    try:
+        sm.cleanup_subscription_files()
+    except SubscriptionManagerError:
+        # Ignore cleanup file removal issues (optional based on policy)
+        pass
+
+    for w in WaitUntil(timeout=timeout, interval=interval):
```

Comment:
`exec_command` by default has a timeout option which is being used by `sm.register`... we shouldn't be using `WaitUntil` logic here.